bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -198,26 +201,26 @@ actual class Scanner internal constructor(
         }
     }
 
-    class ScanOptions {
-        class Builder(
+    class ScanSettings private constructor(val scanOptions: Map<Any?, *>) {
+        data class Builder(
             // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionallowduplicateskey
-            private var allowDuplicateKeys: Boolean = true,
+            var allowDuplicateKeys: Boolean = true,
             // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionsolicitedserviceuuidskey
-            private var solicitedServiceUUIDsKey: List<UUID>? = null,
+            var solicitedServiceUUIDsKey: List<UUID>? = null,
         ) {
```

Comment:
Do these options not have any equivalent setting on Android?

Reply:
I know allowDuplicate keys has [one](https://developer.android.com/reference/android/bluetooth/le/ScanSettings.Builder#setCallbackType(int)) but for the other one I could not figure out the difference between  scan option [solicitedserviceuuidskey](https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionsolicitedserviceuuidskey) and the parameter of [scanForPeripheralsWithServices](https://developer.apple.com/documentation/corebluetooth/cbcentralmanager/1518986-scanforperipheralswithservices) method serviceUUIDs. Also Android does not offer a similar option on the scan settings [builder](https://github.com/NordicSemiconductor/Android-Scanner-Compat-Library/blob/master/scanner/src/main/java/no/nordicsemi/android/support/v18/scanner/ScanSettings.java). Android just takes the UUID filter on the scanForDevices(filter, scanSettings).

Reply:
Well the point is of course to unify the API.

It sounds like this could be crossplatform options, but maybe it's sufficient to create an issue for that for now and this can be a building block.

Reply:
I will create an issue from this as the only option I see common is more or less allowDuplicateKeys so not sure how to unify it.

Reply:
you would `expect` the settings in common, with the `allowDuplicateKeys` included, and then add non-standard options only in `actual`.